github_workflow.md

```diff
@@ -0,0 +1,153 @@
+# GitHub Milestone and Issue Workflow
+
+This purpose of this document is to describe our development workflow on
+GitHub; in particular, how we use milestones and issues. We recognize that our
+many subprojects may use variations of these ideas. This particular workflow is
+based on the practices from the JupyterLab repo
+(https://github.com/jupyterlab/jupyterlab) and are optional for Jupyter
+subprojects to follow.
+
+## Milestones
+
+We use Github milestones to track issues and pull requests related to particular
+releases of a subproject (https://github.com/jupyterlab/jupyterlab/milestones).
+
+* These milestones should be labeled with the version numbers of the releases
+  (`4.3`, `5.0`, etc.).
+* Subprojects should only have a few numbered milestones. Beyond one or two
+  future releases, we have found that detailed milestones become meaningless.
+* Subprojects should also have a `Backlog` milestone for issues and pull
+  requests that have not been assigned to a particular release milestone. This
+  also includes new feature requests ("wishlist") issues that we haven't
+  committed to implementing. Essentially, `Backlog` means "at some point in the
+  future, maybe."
```

Comment:
"With backlog, everything with a numbered milestone is considered a 'blocker' for the release?"

Mostly yes. We still make regular passes at the numbered milestone issues to make sure they don't need to be delayed, but that work of auditing the numbered milestone issues goes really quick as there are fewer of them. As a result, we end up auditing them quite often.

So in a sense we still do those passes where we actively decide to not do certain things.

But yes, one of the main points is that we have to pay very close attention to the backlog, especially each time we cut a release and start to plan for the next one. My guess is that once JupyterLab 1.0 is close to being released we will create a 2.0 milestone and start assigning backlog issues to the 2.0 ms.